test/powershell/Host/TabCompletion/TabCompletion.Tests.ps1

```diff
@@ -768,6 +768,30 @@ ConstructorTestClass(int i, bool b)
         }
     }
 
+    Context 'Help Module parameter completion' {
+        BeforeAll {
+            $utilityModule = 'Microsoft.PowerShell.Utility'
+            $managementModule = 'Microsoft.PowerShell.Management'
+            $allPowerShellModules = ((Get-Module -Name Microsoft.Power* -ListAvailable).Name | Sort-Object -Unique) -join ' '
+        }
```

Comment:
To make the tests more stable I think we should explicitly load module we use in the tests.

Reply:
I feel like this is not needed since all the modules are fetched using wildcard and should be auto loaded in the test before we use `Get-Module -ListAvailable`. We would need to keep the list of module names if we wanted to import modules beforehand explicitly, which gets more complicated because some assemblies are windows only.

Reply:
Best practice is any test shouldn't depend on another test or its order.

Reply:
```pwsh
[-] Error occurred in Context block 0ms
        RuntimeException: The following error occurred while loading the extended type data file:
        Error in TypeData "Microsoft.PowerShell.Commands.GetCounter.PerformanceCounterSampleSet": The member Readings is already present.
        Error in TypeData "Microsoft.PowerShell.Commands.GetCounter.PerformanceCounterSampleSet": The member DefaultDisplayPropertySet is already present.
        Error in TypeData "Microsoft.PowerShell.Commands.GetCounter.PerformanceCounterSample": The member DefaultDisplayPropertySet is already present.
        Error in TypeData "Microsoft.PowerShell.Commands.GetCounter.CounterSet": The member Counter is already present.
        Error in TypeData "System.Diagnostics.Eventing.Reader.ProviderMetadata": The member DefaultDisplayPropertySet is already present.
        Error in TypeData "System.Diagnostics.Eventing.Reader.ProviderMetadata": The member ProviderName is already present.
        Error in TypeData "System.Diagnostics.Eventing.Reader.EventLogRecord": The member DefaultDisplayPropertySet is already present.
        Error in TypeData "System.Diagnostics.Eventing.Reader.EventLogConfiguration": The member DefaultDisplayPropertySet is already present.
        at <ScriptBlock>, C:\Users\armaa\Documents\git-repos\PowerShell\test\powershell\Host\TabCompletion\TabCompletion.Tests.ps1: line 893
        at Invoke-Blocks, C:\Users\armaa\OneDrive\Documents\PowerShell\Modules\Pester\4.10.1\Functions\SetupTeardown.ps1: line 135
        at Invoke-TestGroupSetupBlocks, C:\Users\armaa\OneDrive\Documents\PowerShell\Modules\Pester\4.10.1\Functions\SetupTeardown.ps1: line 121
        at DescribeImpl, C:\Users\armaa\OneDrive\Documents\PowerShell\Modules\Pester\4.10.1\Functions\Describe.ps1: line 209
```

Above is what happens in test if you try to import all microsoft powershell modules.

Reply:
Is it after you added `Import-Module Microsoft.PowerShell.Diagnostics`?

Reply:
Yep that is correct.

Reply:
I included `-ErrorAction SilentlyContinue` so test doesn't fail here. It seems to fail only for diagnostics module.

Reply:
This just shows a side effect of the other tests. It shouldn't be. But that's not the subject of this PR.

Reply:
We can probably raise a separate issue to cover it but agree it shouldn't be fixed in this PR.